sharktank/sharktank/ops/_registry.py

```diff
@@ -386,10 +416,66 @@ def overridable(
     """
     if f is None:
         return functools.partial(
-            overridable, is_trivially_replicable=is_trivially_replicable
+            overridable,
+            dispatch_args=dispatch_args,
+            is_trivially_replicable=is_trivially_replicable,
         )
 
     dispatcher = SignatureDispatcher(f, is_trivially_replicable=is_trivially_replicable)
     functools.update_wrapper(dispatcher, f)
 
+    if dispatch_args is not None:
+        dispatcher.trampoline(make_default_trampoline(f, dispatch_args=dispatch_args))
+
     return dispatcher
+
+
+def make_default_trampoline(
+    f: Callable[..., Any], /, *, dispatch_args: Iterable[int | str]
+) -> Callable[..., Any]:
+    signature = inspect.signature(f)
+    signature_arg_names = list(signature.parameters.keys())
+    dispatch_args = [
+        a if isinstance(a, str) else signature_arg_names[a] for a in dispatch_args
+    ]
+
+    def trampoline(_signature_dispatcher_: SignatureDispatcher, *args, **kwargs) -> Any:
+        # We need the signature created here and not captured from the parent scope.
+        # Otherwise torch tracing fails.
+        signature = inspect.signature(f)
+        bound_args = signature.bind(*args, **kwargs)
+        bound_args.apply_defaults()
```

Comment:
Why can't you remove the call to `bound_args.apply_defaults()` and reimplement it to make it tracing friendly?
Probably the tracing problem is that in `bound_args.apply_defaults()` `bound_args.arguments` gets completely replaced with a new value.

Reply:
I removed apply_defaults and all tests are still passing. I'm not sure if this achieves the full desired behavior, but if you send me PR making the desired changes I'll gladly accept it. I want to build off of this PR, so I want to land this now, though.